### PR TITLE
Optimize layer iteration to prevent performance overhead.

### PR DESCRIPTION
**Pull request by NonSwag**
## Overview

Clamp the min and max y values to the min and max values of the chunk

## Description

Adjusted the layer iteration logic in the Region class to use the minimum and maximum Y-values from the chunk. This prevents unnecessary iterations when getMinimumY or getMaximumY returns extreme values, improving performance efficiency.

```[tasklist]
### Submitter Checklist
- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).
```

### CODE REVIEW

**Comment by NonSwag** [comment]
I was not sure whether it would be more appropriate to use `getMinSectionPosition` or `getMaxSectionPosition` over `getMinY` or `getMaxY`

File: worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java [hunk 1]
```diff
@@ -403,7 +403,9 @@ default IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet set) {
         if (tx >= min.x() && bx <= max.x() && tz >= min.z() && bz <= max.z()) {
             // contains some
             boolean processExtra = false;
-            for (int layer = getMinimumY() >> 4; layer <= getMaximumY() >> 4; layer++) {
+            final int max1 = Math.max(getMinimumY(), chunk.getMinY());
+            final int min1 = Math.min(getMaximumY(), chunk.getMaxY());
+            for (int layer = max1 >> 4; layer <= min1 >> 4; layer++) {
```
Comment (PierreSchwang) [inline worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java:408]: I think, something like

```java
final int minLayer = Math.max(getMinimumY(), chunk.getMinY()) >> 4;
// ...
for (int layer = minLayer; //...
```
would be more readable. It seems like the variable names are mixed up either way

Comment (NonSwag) [reply worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java:408]: Gonna rename them later
Thanks for the input
Btw the names are not mixed up

**Comment by PierreSchwang** [comment]
Would be great if the variables would either be called `minY` instead of `minLayer` or keep them `minLayer` but put the bit shift directly after the Math.min/max instead of inside the loop - just so that the naming fits

**Review by SirYwell — APPROVED** [review]

**Review by PierreSchwang — APPROVED** [review]

**Pull request merged**